clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/lifecycle/LaunchFailureNotificationAgent.java

```diff
@@ -0,0 +1,258 @@
+      .filter(c -> c instanceof NetflixAmazonCredentials)
+      .map(AccountCredentials::getAccountId)
+      .collect(Collectors.toList());
+
+    AmazonSQS amazonSQS = amazonClientProvider.getAmazonSQS(queueARN.account, queueARN.region);
+    this.queueId = ensureQueueExists(amazonSQS, queueId, queueARN, topicARN);
+
+    AmazonSNS amazonSNS = amazonClientProvider.getAmazonSNS(topicARN.account, topicARN.region);
+    this.topicId = ensureTopicExists(amazonSNS, topicId, topicARN, allAccountIds, queueARN);
+
+    AtomicInteger messagesProcessed = new AtomicInteger(0);
+    while (messagesProcessed.get() < properties.getMaxMessagesPerCycle()) {
+      ReceiveMessageResult receiveMessageResult = amazonSQS.receiveMessage(
+        new ReceiveMessageRequest(queueId)
+          .withMaxNumberOfMessages(AWS_MAX_NUMBER_OF_MESSAGES)
+          .withVisibilityTimeout(properties.getVisibilityTimeout())
+          .withWaitTimeSeconds(properties.getWaitTimeSeconds())
+      );
+
+      receiveMessageResult.getMessages().forEach(message -> {
+        log.debug("Received message (queue: {}, message: {})", queueARN.name, message.getBody());
+
+        try {
+          NotificationMessageWrapper notificationMessageWrapper = objectMapper.readValue(
+            message.getBody(), NotificationMessageWrapper.class
+          );
+
+          NotificationMessage notificationMessage = objectMapper.readValue(
+            notificationMessageWrapper.message, NotificationMessage.class
+          );
+
+          if (SUPPORTED_LIFECYCLE_TRANSITION.equalsIgnoreCase(notificationMessage.event)) {
+            handleMessage(notificationMessage);
+          }
+        } catch (IOException e) {
+          log.error("Unable to convert NotificationMessage (body: {})", message.getBody(), e);
+        }
+
+        deleteMessage(amazonSQS, queueId, message);
+        messagesProcessed.incrementAndGet();
+      });
+
+      if (receiveMessageResult.getMessages().size() < AWS_MAX_NUMBER_OF_MESSAGES) {
+        break;
+      }
+    }
+  }
+
+  private void handleMessage(NotificationMessage notificationMessage) {
+    // TODO-AJ This should eventually tag the ASG with an alert
+    log.info(
+      "Failure to launch instance (asgName: {}, reason: {})",
+      notificationMessage.autoScalingGroupName,
+      notificationMessage.statusMessage
+    );
+  }
+
+  /**
+   * Ensure that the topic exists and has a policy granting all accounts permission to publish messages to it
+   */
+  private static String ensureTopicExists(AmazonSNS amazonSNS,
+                                          String topicUrl,
+                                          ARN topicARN,
+                                          List<String> allAccountIds,
+                                          ARN queueARN) {
+    if (topicUrl == null) {
+      try {
+        amazonSNS.getTopicAttributes(topicARN.arn);
+      } catch (Exception e) {
+        amazonSNS.createTopic(topicARN.name).getTopicArn();
+      }
```

Comment:
`createTopic` is idempotent, you could totally skip the call to `getTopicAttributes`.